src/main/java/org/jboss/logmanager/formatters/Formatters.java

```diff
@@ -477,8 +477,7 @@ public static FormatStep dateFormatStep(final TimeZone timeZone, final String fo
             final int minimumWidth,
             final boolean truncateBeginning, final int maximumWidth) {
         return new JustifyingFormatStep(leftJustify, minimumWidth, truncateBeginning, maximumWidth) {
-            final DateTimeFormatter dtf = DateTimeFormatter
-                    .ofPattern(formatString == null ? "yyyy-MM-DD HH:mm:ss,SSS" : formatString);
+            final DateTimeFormatter dtf = StandardDateFormat.resolve(formatString);
```

Comment:
What if instead of the enum, you did something along these lines in (half-Java, half-pseudocode):

```java
private DateTimeFormatter resolve(String formatString) {
    if (formatString.length() > 1 && formatString.charAt(0) == '#') {
        formatString = formatString.substring(1);
        if (formatString.charAt(0) != '#') { // double-## means pass thru
            switch (formatString) {
                case "ISO_DATE": return DateTimeFormatter.ISO_DATE;
                // ... and one case for each known string that we want to allow ...
                default: return DateTimeFormatter.ofPattern("'<Unknown format " + formatString + ">'");
            }
        }
    }
    return DateTimeFormatter.ofPattern(formatString);
}
```

This keeps the code minimal and simple, is guaranteed to be compatible with future `DateTimeFormatter` utilizations of `#`, and is forward-compatible as well in case we want to support fancier things like putting the format name in mid-string or whatever.

Reply:
Do you mean the text format would be something like `%d{#ISO_DATE}`?

Reply:
Exactly. This way we never need to be concerned about syntactic conflicts between constant names and format strings.

Reply:
On the other hand, that character really has _no meaningful_ use from the user standpoint, with the exception of identifying, by convention, date/time pattern formats from constants — something you can achieve performing proper parsing.

I find it more confusing, in the sense that single characters in that particular string (e.g.: `%date{ISO8601} |- %highlight(%5level) in %cyan(%logger{32}:%line) %magenta([%t - %m%n%ex`) have a particular meaning.

Still it's a minor detail, but it amounts.

Reply:
@x80486 The main reason for the prefix character is most characters are reserved for a `DateTimePattern`. While `ISO_DATE` or `ISO08601` is likely not going to bean issue, it could be a valid pattern at some point.